Libraries/LibPthread/pthread.cpp

```diff
@@ -157,70 +167,233 @@ int pthread_sigmask(int how, const sigset_t* set, sigset_t* old_set)
     return 0;
 }
 
-int pthread_mutex_init(pthread_mutex_t* mutex, const pthread_mutexattr_t* attributes)
+#define PTHREAD_FUTEX_UNLOCKED 0u
+#define PTHREAD_FUTEX_LOCKED 1u
+#define PTHREAD_FUTEX_LOCKED_WITH_WAITERS 2u
+
+static int futex_lock(uint32_t& futex_addr, uint32_t futex_flags, const timespec* abstime = nullptr, bool no_wait = false)
 {
-    mutex->lock = 0;
-    mutex->owner = 0;
-    mutex->level = 0;
-    mutex->type = attributes ? attributes->type : PTHREAD_MUTEX_NORMAL;
-    return 0;
+    ASSERT(!(futex_flags & ~(FUTEX_PRIVATE_FLAG | FUTEX_CLOCK_REALTIME)));
+    for (;;) {
+        uint32_t expected = PTHREAD_FUTEX_UNLOCKED;
+        if (AK::atomic_compare_exchange_strong(&futex_addr, expected, PTHREAD_FUTEX_LOCKED, AK::MemoryOrder::memory_order_acquire)) {
+            atomic_thread_fence(AK::MemoryOrder::memory_order_release);
```

Comment:
Why is this needed here?

Reply:
good point, yeah I think we don't need the release here